include/aws/io/private/event_loop_impl.h

```diff
@@ -172,6 +192,8 @@ struct aws_event_loop *aws_event_loop_new_default(struct aws_allocator *alloc, a
 /**
  * Creates an instance of the default event loop implementation for the current architecture and operating system using
  * extendable options.
+ *
+ * Please note the event loop type defined in the options will be ignored.
  */
 AWS_IO_API
 struct aws_event_loop *aws_event_loop_new_default_with_options(
```

Comment:
Can we just eliminate all these alternate constructors and replace them with a single `aws_event_loop_new(allocator, options)`?

I mean ... they were already in a private header
or is that going to happen in some future PR?

Reply:
I eventually keep two constructors: `aws_event_loop_new_default` and `aws_event_loop_new`. `aws_event_loop_new_default` was used cross the aws-c-io tests to create default event loop. If we remove it, we probably need create help functions in all those test files, therefore I leave it there...